DAOS-12370 ci: Functional test on Ubuntu

brianjmurrell opened the pull request:
### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

brianjmurrell committed:
DAOS-12370 ci: Functional test on Ubuntu

Required-githooks: true


# NOTE: The following default set of pragmas will result in no build
#       or test.
#
#       Please edit the list of pragmas to produce the minimal amount of build
#       and test to prove your PR works.
#
#       Or if you have already done the above, and your PR is ready for it's
#       final for-landing run, remove all of the following pragmas.
#
#       More detail can be found at
#       https://wiki.hpdd.intel.com/display/CI/Commit+Pragmas.
Skip-checkpatch: true
Skip-python-bandit: true
Skip-build: true
# Or you can skip specific PR build stages:
#Skip-build-ubuntu20-rpm: true
#Skip-build-el8-rpm: true
#Skip-build-leap15-rpm: true
#Skip-build-el8-gcc: true
#Skip-build-el8-gcc-debug: true
#Skip-build-el8-gcc-release: true
#Skip-build-leap15-gcc: true
#Skip-build-leap15-icc: true
#Skip-build-ubuntu-clang: true

# And/or you could just enable quick-build below which reduces build tims considerably
Quick-build: true
# Just build and run Functional tests, skipping all others
Quick-Functional: true
# or building with multiple jobs
#Parallel-build: true

Skip-unit-tests: true
# Or you could choose which Unit Test stages to skip below
#Skip-nlt: true
#Skip-unit-test: true
#Skip-unit-test-memcheck: true
Skip-test: true

# Every run should include the features it is testing.  Uncomment the below
# line and add the features (space separated) your PR should be testing for.
# I.e. rebuild, ...
#Features: 

# Or you can choose which test stages to skip below
#Skip-coverity-test: true
# Skip all Functional test stages
#Skip-func-test: true
# Skip all Functional test stages on VMs
#Skip-func-test-vm: true
# Don't skip the valgrind functional test
#Skip-func-test-vm-valgrind: false
# Skip Functional test stages on EL7, EL8 or Leap 15
#Skip-func-test-el7: true
#Skip-func-test-el8: true
#Skip-func-test-leap15: true
# Skip all Functional test stages on hardware
#Skip-func-hw-test: true
# or just specific hardware test stages
#Skip-func-hw-test-small: true
#Skip-func-hw-test-medium: true
#Skip-func-hw-test-large: true
#Skip-scan-rpms: true
#Skip-test-rpms: true
# or per distro variants:
#Skip-scan-centos-rpms: true
#Skip-scan-centos-7-rpms: true
#Skip-scan-centos-8-rpms: true
#Skip-scan-leap-15-rpms: true
#Skip-test-centos-rpms: true
#Skip-test-centos-7-rpms: true
#Skip-test-centos-8-rpms: true
#Skip-test-centos-8.3-rpms: true

# If you do run any of the above Functional test stages, please use the following to
# limit your tests run to the tests/features you are working on
# You can add the tags for your tests to the appropriate pragmas below:
#Test-tag: -hw
#Test-tag-hw-small: hw,small
#Test-tag-hw-medium: hw,medium,ib2
#Test-tag-hw-large: hw,large

# If you want to allow Hardware testing to run even if VM testing fails:
Allow-unstable-test: true

# if you want to test your PR with a PR of a component, you can speify that PR:
#PR-repos: project@PR-number[:build_num]

# if you want to disable using RPMs in the Functional tests:
#RPM-test: false

# Or if you want to test with daos RPMs that are already built (i.e. in another PR or in the repo):
#RPM-test-version: version[-release]
# to get the latest RPMs for the current verison:
#RPM-test-version: 2.3.101

# This should be figured out automatically, but you can force it with
#Doc-only: true

# If you'd prefer not to have your PRs cluttered up with stage failure
# comments, uncomment this
# Skip-PR-comments: true

# ansible-role-snapshot_host-branch: branch-name
# ansible-role-kvm-branch: branch-name

# to use a different cluster for various stages
# VM1-label: ci_vm1
# Ubuntu-VM9-label: ci_vm9
# Leap15-VM9-label: ci_vm9
# EL8-VM9-label: ci_vm9

# to tell CI not to skip a test due to a ticket
# Fixes: DAOS-1234
#
# If you are ready to push a full CI build and test you can uncomment
# the below pragma:
#Full-CI-run: true
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Use pipeline-lib branch

PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

phender commented on src/tests/ftest/launch.py:
```diff
@@ -3171,7 +3171,6 @@ def _rename_avocado_test_dir(self, test, jenkinslog):
 
         """
         avocado_logs_dir = self.avocado.get_logs_dir()
-        logger.info("avocado_logs_dir = " + avocado_logs_dir)
```
Why was this removed?  This is useful information.

phender commented on src/tests/ftest/launch.py:
```diff
@@ -428,6 +433,14 @@ def get_logs_dir(self):
 
         """
         default_base_dir = os.path.join("~", "avocado", "job-results")
+        logger.info("default_base_dir = %s", default_base_dir)
+        datadir_paths_logs_dir = self.get_setting("datadir.paths", "logs_dir", default_base_dir)
+        logger.info("datadir.paths = %s ", datadir_paths_logs_dir)
+        time.sleep(2)
+        datadir_paths_logs_dir = self.get_setting("datadir.paths", "logs_dir", default_base_dir)
+        logger.info("datadir.paths = %s ", datadir_paths_logs_dir)
```
If we want to keep these log messages, they should not be info, but debug to avoid cluttering the console:
```suggestion
        logger.debug("default_base_dir = %s", default_base_dir)
        datadir_paths_logs_dir = self.get_setting("datadir.paths", "logs_dir", default_base_dir)
        logger.debug("datadir.paths = %s ", datadir_paths_logs_dir)
        time.sleep(2)
        datadir_paths_logs_dir = self.get_setting("datadir.paths", "logs_dir", default_base_dir)
        logger.debug("datadir.paths = %s ", datadir_paths_logs_dir)
```

brianjmurrell replied:
This is all just part of the process of getting a new distro to run in CI.  Lots of debug code that will ultimately be removed.

phender commented on src/tests/ftest/launch.py:
```diff
@@ -428,6 +433,14 @@ def get_logs_dir(self):
+        time.sleep(2)
+        datadir_paths_logs_dir = self.get_setting("datadir.paths", "logs_dir", default_base_dir)
+        logger.info("datadir.paths = %s ", datadir_paths_logs_dir)
+        if datadir_paths_logs_dir != '/var/tmp/ftest/avocado/job-results':
+            sys.exit(-1)
```
This is debug code, right?  This check probably doesn't make sense outside of our CI environment.

phender commented on src/tests/ftest/util/apricot/apricot/test.py:
```diff
@@ -507,7 +509,10 @@ def setUp(self):
         self.fault_injection.start(self.params.get("fault_list", '/run/faults/*'), self.test_dir)
 
-        self.context = DaosContext(self.prefix + '/lib64/')
+        self.context = DaosContext(os.path.join(self.prefix,
+                                                'lib', 'x86_64-linux-gnu'
+                                                if detect().name.lower() == 'ubuntu'
+                                                else 'lib64'))
```
The if statement only acts on the last argument:
```
>>> os.path.join('/', 'lib', 'x86_64-linux-gnu' if False else 'lib64')
'/lib/lib64'
```
I suspect you want this behavior:
```
>>> os.path.join('/', os.path.join('lib', 'x86_64-linux-gnu') if False else 'lib64')
'/lib64'
>>> os.path.join('/', os.path.join('lib', 'x86_64-linux-gnu') if True else 'lib64')
'/lib/x86_64-linux-gnu'
```
```suggestion
        lib_dir = os.path.join('lib', 'x86_64-linux-gnu') if detect().name.lower() == 'ubuntu' else 'lib64'
        self.context = DaosContext(os.path.join(self.prefix, lib_dir))
```

brianjmurrell replied:
Except that the two path choices are `/lib/x86_64-linux-gnu` and `/usr/lib64`.  I'll take another look at this code and see if it is currently doing as I am hoping.  Thanks for pointing out that there may be issues here though.

phender commented on src/tests/ftest/util/apricot/apricot/test.py:
```diff
@@ -195,17 +195,15 @@ def cancel_for_ticket(ticket, skip_list):
-                                          "skipped", ticket)
-                            return
+                    if ticket in open(os.path.join(os.sep, 'tmp',
+                                                   'commit_fixes')).read().splitlines():
```
```suggestion
                    with open(os.path.join(os.sep, 'tmp',
                                           'commit_title')) as commit_handle:
                        if ticket in commit_handle.read().splitlines():
```

brianjmurrell committed:
Fix version

PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild daos@PR-11205:92
RPM-test-version: 2.3.103
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Add mpifileutils-mpich Depends

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

brianjmurrell committed:
Add lbzip2 D:

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

brianjmurrell committed:
Move mpifileutils to tests-internal

As it should not be an external dependency.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

brianjmurrell committed:
Fix mpifileutils package name

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

brianjmurrell committed:
Install daos-tests-internal...

Instead of daos-client-tests-openmpi for functional testing.

Required-githooks: true

PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild daos@PR-11205:97
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Need mpich; nothing depends on it

Required-githooks: true

PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Bit o' debug

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

brianjmurrell committed:
More support for Debian alternatives...

For finding an MPI stack.

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

brianjmurrell committed:
DO NOT LAND

Debug for the above

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

brianjmurrell committed:
Fix IOR dependency

PR-repos-ubuntu20: spdk@PR-58:lastBuild raft@PR-66 mpich@PR-64:lastBuild hdf5@PR-38:lastBuild ior@PR-10:lastBuild mpifileutils-pkg@PR-34:lastBuild dtcmp@PR-6:lastBuild lwgrp@PR-6:lastBuild libcircle@PR-6:lastBuild hdf5-vol-daos@PR-38:lastBuild
Skip-PR-comments: true
Quick-build: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-el8-gcc: true
Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-test-el8: true
Skip-func-test-ubuntu20: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Ubuntu-VM9-label: stage_vm9
Test-tag: test_dm_preserve_props_fs_copy_posix_dfs test_dm_serial_small_dserialize test_llnl test_cart_iv
Fixes: DAOS-12427

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
